code/game/objects/structures.dm

```diff
@@ -160,27 +212,41 @@
 				break
 		if(!success)
 			for(var/obj/O in T)
-				for(var/b_type in blend_objects)
+				for(var/b_type in GLOB.default_blend_objects)
 					if(istype(O, b_type))
 						success = 1
 						for(var/obj/structure/S in T)
 							if(istype(S, src))
 								success = 0
-						for(var/nb_type in noblend_objects)
+						for(var/nb_type in GLOB.default_noblend_objects)
 							if(istype(O, nb_type))
 								success = 0
```

Comment:
Suggest writing a helper for the above:
```
/obj/structure/proc/can_blend_with(atom/A)
	if(is_type_in_list(A, GLOB.default_noblend_objects))
		return FALSE
	if(istype(A, type)
		return FALSE // Why does it make this check?
	if(is_type_in_list(A, GLOB.default_blend_objects))
		return TRUE
	return FALSE
```
One of the worst things about this code currently is that the above logic is incomprehensible and copy-pasted in several places.
It's not obvious to me why the blend type lists should be global.

Reply:
Moved this into a helper, but not quite as granular as this one.